cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -134,9 +134,9 @@ int64_t getBatchNbytes(const arrow::RecordBatch& rb) {
 }
 
 std::shared_ptr<arrow::Array> makeNullBinaryArray(std::shared_ptr<arrow::DataType> type, ShuffleBufferPool* pool) {
-  std::shared_ptr<arrow::Buffer> offsetBuffer;
+  std::shared_ptr<arrow::ResizableBuffer> offsetBuffer;
   size_t sizeofBinaryOffset = sizeof(arrow::LargeStringType::offset_type);
-  GLUTEN_THROW_NOT_OK(pool->allocate(offsetBuffer, sizeofBinaryOffset * 2));
+  GLUTEN_THROW_NOT_OK(pool->allocateDirectly(offsetBuffer, sizeofBinaryOffset * 2));
```

Comment:
@jinchengchenghh Change to `allocateDirectly` here to avoid allocating compression buffers from combined buffer.